### ISSUE TITLE
Standard Library documentation build fails when optional libraries are disabled

### ISSUE THREAD

**Issue by yurivict** [ocaml/ocaml#10584]
When built without threads it fails:
```
File "../otherlibs/systhreads/condition.mli", line 41, characters 16-23:
41 | m] atomically unlocks the mutex [m] and suspends the
                     ^^^^^^^
Error: Unbound module Mutex
```

configure arguments: ```-verbose -prefix "/usr/local"  --disable-force-safe-string  CC="cc"  ASPP="cc -c"  PARTIALLD="ld -r" AS="as " --disable-systhreads```

Version: 4.12.0
OS: FreeBSD 13

**Comment by dra27** [comment]
Workaround is to add `--disable-stdlib-manpages` to the configure call (which obviously only helps if you don't actually need the man pages for the standard library)

**Comment by dra27** [comment]
I've opened a fix for this for 4.13, but the fix is different for 4.12 and earlier as that part of the build system has been refactored. Do you have a strong need for this to be fixed in 4.12.1? (it has been broken like this since 4.08.0)

**Comment by yurivict** [comment]
> Do you have a strong need for this to be fixed in 4.12.1?

No.

**Comment by dra27** [comment]
It’ll be fixed in the next beta of 4.13 - thanks for the report!
